lib/decorateWithMetrics.js

```diff
@@ -136,55 +137,83 @@ function toDuration(millis, raw = false) {
+ * @param {Object} [options]
+ * @param {boolean} [options.raw=false] - setting this to true will print raw instead of human friendly times
  * Useful when you copy the printed table to excel and let excel do the number formatting
+ * @param {'console'|'csv'|'json'|'string'} [options.format=console] The output format of this function
+ * `string` is useful when __DEV__ is set to `false` as writing to the console is disabled, but the result of this
+ * method would still get printed as output
+ * @returns {string|undefined}
  */
-function printMetrics(raw = false) {
-    const {totalTime, summaries, lastCompleteCall = {endTime: -1}} = getMetrics();
+function printMetrics({raw = false, format = 'console'} = {}) {
+    const {totalTime, summaries, lastCompleteCall} = getMetrics();
 
-    const prettyData = _.chain(summaries)
+    const tableSummary = MDTable.factory({
+        heading: ['method', 'total time spent', 'max', 'min', 'avg', 'time last call completed', 'calls made'],
+        leftAlignedCols: [0],
+    });
+
+    const methodCallTables = _.chain(summaries)
         .filter(method => method.avg > 0)
         .sortBy('avg')
         .reverse()
-        .map(({
-            calls, methodName, lastCall, ...summary
-        }) => {
-            const prettyTimes = _.chain(summary)
-                .map((value, key) => ([key, toDuration(value, raw)]))
-                .object()
-                .value();
-
-            const prettyCalls = calls.map(call => ({
-                startTime: toDuration(call.startTime, raw),
-                endTime: toDuration(call.endTime, raw),
-                duration: toDuration(call.duration, raw),
-                args: JSON.stringify(call.args)
-            }));
-
-            return {
+        .map(({methodName, calls, ...methodStats}) => {
+            tableSummary.addRow(
                 methodName,
-                ...prettyTimes,
-                'time last call completed': toDuration(lastCall.endTime, raw),
-                calls: calls.length,
-                prettyCalls,
-            };
+                toDuration(methodStats.total, raw),
+                toDuration(methodStats.max, raw),
+                toDuration(methodStats.min, raw),
+                toDuration(methodStats.avg, raw),
+                toDuration(methodStats.lastCall.endTime, raw),
+                calls.length,
+            );
+
+            const callsTable = MDTable.factory({
+                title: methodName,
+                heading: ['start time', 'end time', 'duration', 'args'],
+                leftAlignedCols: [3],
+                rows: calls.map(call => ([
+                    toDuration(call.startTime, raw),
+                    toDuration(call.endTime, raw),
+                    toDuration(call.duration, raw),
+                    call.args.map(String).join(', ').slice(0, 60), // Restrict cell width to 60 chars max
```

Comment:
We're limiting this as it breaks the table when it's too large 
I find it better to not JSON.stringify the data and instead keep it simple like: `account, [object Object]` 
otherwise it adds too much noise